### ISSUE TITLE
Icons of applications stored in adoptable storage dissappear after system reboot

### ISSUE THREAD

**Issue by gitter77** [Deletescape-Media/Lawnchair#734]
## Description

I recently moved some apps to adoptable storage (android built in feature) but after each system reboot their respective icons dissappears from the home launcher screen and they can only be found in drawer.
I have no problem with other apps (stored in internal storage).
I noticed a toast showing "**failed to get app filter**"
I have tried resetting settings, a reinstallation but no luck.
Out of subject : Hope you can fix this in feature release as i like this launcher a lot especially with the blur option and please try to keep it less google dependent as you can, there is plenty of other launchers derived from pixel launcher out there which "google fans" can use and have fan with (that being said because i saw many  features requests trying to transform this nice and clean launcher to a some sort of a google product which i am not interested in)
Thanks

## Expected Behavior

All home launcher icons should persist after a system reboot

## Actual Behavior

Icons of specific apps ( the ones moved to adoptable storage ) disappears from home launcher after a system reboot

## Environment

* Device:
* Android version: 6.0.1
* Launcher version: 1.0.1063
* Rom:cm13

**Comment by AxiK101** [comment]
That also happens in original pixel launcher. Old story but the only way to bypass it is to move all apps in internal storage .

**Comment by divadsn** [comment]
Sorry, but this issue is on nearly any launcher app out there, apps which are on SD card (no matter if adopted or not) will show up after the SD card has been initialized.

**Comment by fonix232** [comment]
@divadsn would it be possible to catch the SD card init event and reload the app list/icons when that happens?

**Comment by divadsn** [comment]
@fonix232 afaik there is a system broadcast sent when the SD card got initialized, but not when apps got loaded into the system, which happens slowly depending on the SD card speed.

**Comment by TheShark27** [comment]
I've moved a lot of apps to the SDcard, I have the same issue but I can easily bypass it.
I've created a shortcut for the "Home" section in Settings. When I have to reboot, I tap it, I change launcher, I select the Pixel launcher (or any other launcher) and then I power off the phone. After booting it up, I wait some minutes so it can load all the apps, then I change my launcher back to LawnChair.
It's a bit slow, but it allows you to move apps to the SDcard, a feature that I love.
